signer-cloud-server/src/test/java/com/wultra/signercloud/server/IntTestUtils.java

```diff
@@ -0,0 +1,315 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.signercloud.server;
+
+import eu.europa.esig.dss.enumerations.SignatureAlgorithm;
+import eu.europa.esig.dss.enumerations.SignatureLevel;
+import eu.europa.esig.dss.model.InMemoryDocument;
+import eu.europa.esig.dss.model.SignatureValue;
+import eu.europa.esig.dss.model.x509.CertificateToken;
+import eu.europa.esig.dss.pades.PAdESSignatureParameters;
+import eu.europa.esig.dss.pades.signature.PAdESService;
+import eu.europa.esig.dss.spi.validation.CommonCertificateVerifier;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequestBuilder;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.StringWriter;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.*;
+import java.security.cert.X509Certificate;
+import java.security.spec.ECGenParameterSpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * Class for generating integration test resources.
+ *
+ * @author Michal Rozehnal, michal.rozehnal@wultra.com
+ */
+public final class IntTestUtils {
+
+    private static final X500Name ROOT_SUBJECT = new X500Name("CN=Root CA, O=Wultra, C=CZ");
+    private static final X500Name INTERMEDIATE_SUBJECT = new X500Name("CN=Intermediate CA, O=Wultra, C=CZ");
+    private static final X500Name USER_SUBJECT = new X500Name("CN=User CA, O=Wultra, C=CZ");
+
+    private static IntTestResources testResources;
+
+    /**
+     * Generates and stores resources (keystore, CSR, signed documents) needed for integration tests.
+     *
+     * @return generated resources
+     * @throws Exception when generation or storing of resources fails
+     */
+    public static IntTestResources prepare() throws Exception {
+        if (testResources == null) {
+            createResources();
+        }
+
+        return testResources;
+    }
+
+    private static void createResources() throws Exception {
+        Security.addProvider(new BouncyCastleProvider());
```

Comment:
🤔 This is already in `SignerCloudServerApplication` but not in the static constructor. Maybe we may move that.